keanu-project/src/main/java/io/improbable/keanu/vertices/bool/BoolVertex.java

```diff
@@ -17,19 +18,23 @@
 public abstract class BoolVertex extends DiscreteVertex<BooleanTensor> {
 
     @SafeVarargs
-    public final io.improbable.keanu.vertices.bool.BoolVertex or(Vertex<BooleanTensor>... those) {
+    public final BoolVertex or(Vertex<BooleanTensor>... those) {
         if (those.length == 0) return this;
         if (those.length == 1) return new OrBinaryVertex(this, those[0]);
         return new OrMultipleVertex(inputList(those));
     }
 
     @SafeVarargs
-    public final io.improbable.keanu.vertices.bool.BoolVertex and(Vertex<BooleanTensor>... those) {
+    public final BoolVertex and(Vertex<BooleanTensor>... those) {
         if (those.length == 0) return this;
         if (those.length == 1) return new AndBinaryVertex(this, those[0]);
         return new AndMultipleVertex(inputList(those));
     }
 
+    public static final BoolVertex not(Vertex<BooleanTensor> vertex) {
+        return new NotVertex(vertex);
+    }
```

Comment:
If you fancy the challenge, want to extend this to work for multiple vertices? So we can do something like...

```
public static final BoolVertex not(Vertex<BooleanTensor>... those) {
   if (those.length == 0) // do nothing
   if (those.length == 1) return new NotVertex(those[0])
   return new NotMultipleVertex(inputList(those))
}
```

Reply:
The not multiple would have multiple outputs?

Reply:
Eesh, good point, you can't collapse a not can you. Never mind!